demos/projects/NXP/mimxrt1060/CMakeLists.txt

```diff
@@ -1,17 +1,27 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # SPDX-License-Identifier: MIT
 
-set(MCUX_SDK_PROJECT_NAME ${PROJECT_NAME})
+# set parent scope path
+set(BOARD_DEMO_CONFIG_PATH ${CMAKE_CURRENT_SOURCE_DIR}/config CACHE INTERNAL "Config path")
+set(BOARD_DEMO_FREERTOS_PORT_PATH ${FreeRTOS_ARM_CM4F_PATH} CACHE INTERNAL "FreeRTOS Port used ")
+
+set(MCUX_SDK_PROJECT_NAME mcux-sdk-lib)
+
+add_library(${MCUX_SDK_PROJECT_NAME})
+target_include_directories(${MCUX_SDK_PROJECT_NAME}
+  PUBLIC
+    ${BOARD_DEMO_FREERTOS_PORT_PATH}
+)
+target_link_libraries(${MCUX_SDK_PROJECT_NAME}
+  PUBLIC
+    FreeRTOS
+)
```

Comment:
curious which part of their sdk require this depenency ?

I was under impression that our final binary requires OS depenency and middleware

Reply:
https://github.com/Azure-Samples/iot-middleware-freertos-samples/runs/2933684299#step:5:3647
Here's the build failure